0020-bluetooth-payment-protocol/0020-bluetooth-payment-protocol.md

```diff
@@ -0,0 +1,117 @@
+2. Device manufacturer signs the device public key with their own key pair.
+3. Customer submits the device public key and manufacturer's signature to their ledger operator to authorize the device to spend funds from their account.
+4. If the device manufacturer is trusted to produce secure devices, the ledger operator signs the device public key and a per-payment spending limit with their own key.
+5. Customer stores the ledger operator's signature, certificate, per-payment and total spending limit on the device (**TODO**: would it make more sense to have the manufacturer include the ledger signature and cert in the device?).
+
+## Payment Flow
+
+**[MERC]** indicates steps taken by the Merchant
+
+**[CUST]** indicates steps taken by the Customer's device
+
+1. **[MERC]** Listen to Broadcasts of the [ILP Account Information](#ILPAccountInformation). List customers.
+2. **[MERC]** Select customer for payment, and initiate a connection over BLE.
+3. **[MERC]** Generate IPR with __account__ and __amount__.
+4. **[MERC]** Sign the IPR using their Ed25519 keypair
+5. **[MERC]** Write to:
+  1. `:ipr` with IPR
+  2. `:iprSignature` with Signature of IPR
+  3. `:merchantPublicKey` with the Merchant Public Key 
+  4. `:merchantCertificate` with the Certificate signed by a Certificate Authority
+6. **[MERC]** Write __sign__ to `:action`, to indicate data is ready to be validated/signed.
+7. **[MERC]** Listen to notifications on `:status`
+8. **[CUST]**  Write __"authenticating"__ to `:status`
+9. **[CUST]** Validates payload on Characteristics. If:
+  1. [CA is NOT trusted:](#) write __"invalid"__ to `:status`. Close connection
+  2. [Payment NOT within limits:](#) write __"invalid"__ to `:status`. Close connection
+  3. [Certificate NOT valid for ILP address in IPR:](#) write __"invalid"__ to `:status`. Close connection
+  4. [Merchant signature is NOT on IPR:](#) write __"invalid"__ to `:status`. Close connection
+  5.  (**TODO**: what if the device is super simple and low-power, like a tapFi, and can't easily verify a chain of signatures? Is there any way to make it safe to skip this step?).
+10. **[CUST]**  Wait for Customer authorization (passkey/rhythm/Harry Potter wand...). If:
+  1. [Password is NOT valid:](#) write __"unauthenticated"__ to `:status`. Close connection.
+  2. [Authentication timeout:](#) write __"unauthenticated"__ to `:status`. Close connection.
+  3. [If payment is canceled:](#) write __"canceled"__ to `:status`. Close connection.
+11. **[CUST]** Sign IPR using its ED25519 keypair
+  1. (**TODO**: should it include the merchant cert and/or signature on the IPR in the data being signed?)
+12. **[CUST]** Writes to:
+  1. `:iprAuthorization` with signed IPR, Redemption URI, public key (**TODO**: can the pubkey be recovered from the signature?) and certificate signed by the customer's ledger (and possibly another CA) to the merchant
```

Comment:
If multiple values are being written to the same field, how do you tell them apart?